uniforce/lib/Cleantalk/USP/Security/Firewall.php

```diff
@@ -166,43 +166,58 @@ public function run() {
 	 * @return array Single element array of result
 	 */
 	private function prioritize( $results ){
-		
-		$current_fw_result_priority = 0;
-		$result = array(
-			'status' => 'PASS',
-			'ip' => '',
-		);
-		
-		if( is_array( $results ) ) {
-			foreach ( $results as $fw_result ) {
-				$priority = array_search( $fw_result['status'], $this->statuses_priority ) + ( isset($fw_result['is_personal']) && $fw_result['is_personal'] ? count ( $this->statuses_priority ) : 0 );
-				if( $priority >= $current_fw_result_priority ){
-					$current_fw_result_priority = $priority;
+
+        $final_priority = 0;
+        $result = array(
+            'status' => 'PASS',
+            'ip' => '',
+        );
+
+        if ( is_array($results) ) {
+            foreach ( $results as $fw_result ) {
+                /**
+                 * Calculating priority. Records that have status PASS_BY_TRUSTED_NETWORK gain hardcoded highest priority.
+                 * Personal records are next by priority in accordance of this->statuses_priority.
+                 */
+                $status_priority_from_table = array_search($fw_result['status'], $this->statuses_priority);
+                $is_personal_flag = isset($fw_result['is_personal']) && $fw_result['is_personal'];
+                $is_trusted_network_flag = isset($fw_result['status']) && $fw_result['status'] == 'PASS_BY_TRUSTED_NETWORK';
```

Comment:
Maybe if the status PASS_BY_TRUSTED_NETWORK is found, immediately return the result?

Reply:
Prioritize logic is cloned from the WordPress plugin, I presume we would better keep this as is.